packages/block-library/src/navigation/edit/index.js

```diff
@@ -495,6 +528,13 @@ function Navigation( {
 											setRef( id );
 											onClose();
 										} }
+										onSelectClassicMenu={ ( menu ) => {
+											onClose();
+											convertClassicMenuToBlocks(
```

Comment:
By passing the conversion code down into the dropdown we avoid errors thrown when the dropdown unmounts and an async request triggered by that component resolves. We do this by handling in the parent (which is not unmounted when the option is clicked).